ckanext/datajson/blueprint.py

```diff
@@ -101,137 +113,134 @@ def generate_output(fmt='json', org_id=None):
-                # logger.debug("processing %s" % (pkg.get('title')))
-                extras = dict([(x['key'], x['value']) for x in pkg.get('extras', {})])
-
-                # unredacted = all non-draft datasets (public + private)
-                # redacted = public-only, non-draft datasets
-                if export_type in ['unredacted', 'redacted']:
-                    if 'Draft' == extras.get('publishing_status'):
-                        # publisher = detect_publisher(extras)
-                        # logger.warn("Dataset id=[%s], title=[%s], organization=[%s] omitted (%s)\n",
-                        #             pkg.get('id'), pkg.get('title'), publisher,
-                        #             'publishing_status: Draft')
-                        # _errors_json.append(OrderedDict([
-                        #     ('id', pkg.get('id')),
-                        #     ('name', pkg.get('name')),
-                        #     ('title', pkg.get('title')),
-                        #     ('errors', [(
-                        #         'publishing_status: Draft',
-                        #         [
-                        #             'publishing_status: Draft'
-                        #         ]
-                        #     )])
-                        # ]))
-
-                        continue
-                        # if ('redacted' == export_type and
-                        #         re.match(r'[Nn]on-public', extras.get('public_access_level'))):
-                        #     continue
-                # draft = all draft-only datasets
-                elif 'draft' == export_type:
-                    if 'publishing_status' not in list(extras.keys()) or extras.get('publishing_status') != 'Draft':
-                        continue
-
-                redaction_enabled = ('redacted' == export_type)
-                datajson_entry = Package2Pod.convert_package(pkg, json_export_map, redaction_enabled)
-                errors = None
-                if 'errors' in list(datajson_entry.keys()):
-                    errors_json.append(datajson_entry)
-                    errors = datajson_entry.get('errors')
-                    datajson_entry = None
-
-                if datajson_entry and \
-                        (not json_export_map.get('validation_enabled') or is_valid(datajson_entry)):
-                    # logger.debug("writing to json: %s" % (pkg.get('title')))
-                    output.append(datajson_entry)
-                else:
-                    publisher = detect_publisher(extras)
-                    if errors:
-                        logger.warn("Dataset id=[%s], title=[%s], organization=[%s] omitted, reason below:\n\t%s\n",
-                                    pkg.get('id', None), pkg.get('title', None), publisher, errors)
+        while True:
+            packages = get_datasets(page=page, owner_org=owner_org, with_private=with_private)
+            if packages:
```

Comment:
Coding tip (not a blocker): if you can remove "depth" of logic, do so. This could have been written:
```
if not packages:
  break
[rest of logic]
```
This makes clear WHERE you are breaking and why. It's much easier to read. We aren't trained to do this when doing code 101, but it makes it much cleaner.

Reply:
changed logic flow

Reply:
changed logic flow